sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/DefaultS3ClientBuilderFactory.java

```diff
@@ -38,13 +42,17 @@ public AmazonS3ClientBuilder createBuilder(S3Options s3Options) {
       builder = builder.withClientConfiguration(s3Options.getClientConfiguration());
     }
 
-    if (Strings.isNullOrEmpty(s3Options.getAwsServiceEndpoint())) {
-      builder = builder.withRegion(s3Options.getAwsRegion());
-    } else {
+    if (!Strings.isNullOrEmpty(s3Options.getAwsServiceEndpoint())) {
       builder =
           builder.withEndpointConfiguration(
               new AwsClientBuilder.EndpointConfiguration(
                   s3Options.getAwsServiceEndpoint(), s3Options.getAwsRegion()));
+    } else if (!Strings.isNullOrEmpty(s3Options.getAwsRegion())) {
+      builder = builder.withRegion(s3Options.getAwsRegion());
+    } else {
+      LOG.info(
```

Comment:
This is technically fine - the AWS SDK can figure out the right region if you're running inside EC2. I'm leaving the warning there to not change behavior.